### PR TITLE
3 add runner script

### PR DESCRIPTION
**Pull request by MaHaWo**
### New things
- add `Runner` class that bundles audio recorder together with it's set up and manages for how long data should be recorded. 
- add integration test and unit test for runner. 
- new fixture for folders
- converged on structure for install -> have configs in `user_config_dir/iSparrowRecord`, i.e., in `/HOME/.config/iSparrowRecord`.  iSparrow itself should behave the same way when it progressed to that stage.
Installation configs can be stored there and be retrieved and put together with parameters to have a full setup documentation as a yml file.

### Refactoring
- a few minor changes in the audiorecorder 
- better `__init__.py` 
- `set_up_sparrow.py` is there for setting up a mock sparrow folder structure, lives in `tests`. 
- sorted out default configuration file.

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
-  home: ~/iSparrow 
-  data: ~/iSparrow_data
-  output: ~/iSparrow_output
-Data:
-  Recording:
-    sample_rate: 48000
-    length_s: 3
-    channels: 1
-    mode: "record"
+Output:
```
Comment (MaHaWo) [inline config/default.yml:1]: Removed superfluous confignodes that only iSparrow itself needs.

File: config/custom_example.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Output:
```
Comment (MaHaWo) [inline config/custom_example.yml:1]: Example file, will be used later to show how to change parameters

File: src/iSparrowRecord/audio_recording.py [hunk 3]
```diff
@@ -119,7 +120,7 @@ def __init__(
                     "Output folder for recording object cannot be None in 'record' mode"
                 )
 
-            self.filename_format = "%y%m%d_%H%M%S.wav"
+            self.filename_format = "%y%m%d_%H%M%S"
```
Comment (MaHaWo) [inline src/iSparrowRecord/audio_recording.py:123]: ... this should really only be the format string, not the file type....

File: tests/conftest.py [hunk 4]
```diff
@@ -1,113 +1,56 @@
...
-import pyaudio
+import set_up_sparrow as spf
 
 HOME = None
 DATA = None
 OUTPUT = None
-
-
-# README: the below will later land in setup.py...
-def read_yaml(path: str):
```
Comment (MaHaWo) [inline tests/conftest.py:13]: moved into `set_up.py`.

File: tests/conftest.py [hunk 5]
```diff
@@ -1,113 +1,56 @@
...
-    for p in [ish, isd, iso]:
-        p.mkdir(parents=True, exist_ok=True)
-
-    return ish, isd, iso
+CONFIG = None
 
 
 # add a fixture with session scope that emulates the result of a later to-be-implemented-install-routine
-@pytest.fixture(scope="session", autouse=True)
+@pytest.fixture(scope="module", autouse=True)
```
Comment (MaHaWo) [inline tests/conftest.py:18]: have a new install for each tested module is a bit better a compromise between redoing work and possible interactions than allowing session wide test interactions.

File: tests/conftest.py [hunk 6]
```diff
@@ -1,113 +1,56 @@
...
 
     request.addfinalizer(teardown)
 
 
-@pytest.fixture(scope="session")
+@pytest.fixture(scope="module")
 def folders():
-    global HOME, DATA, OUTPUT
-    return str(HOME), str(DATA), str(OUTPUT)
+    custom_cfgdir = Path(__file__).resolve().parent.parent / Path("config")
```
Comment (MaHaWo) [inline tests/conftest.py:44]: ... needed for runner test

**Comment by MaHaWo** [comment]
@iulusoy If you have time, this one could be reviewed. thanks.

**Review by iulusoy — APPROVED** [review]
Looks good to me. The only thing I would make sure of is that the folder creation and file writing into the global `cfg_dirs` during testing does not interfere with any files that are there from an actual run. It was not clear to me if this is the case.

**Comment by MaHaWo** [comment]
I added a variable that should fix the tests <-> production interference issue. Wanted to test the system in its `natural habitat`, therefore now tmp paths used.

**Pull request merged**